core/templates/services/platform-feature.service.ts

```diff
@@ -0,0 +1,305 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview A service for retriving feature flags - boolean parameters
+ * that are used to determine if features should be enabled.
+ *
+ * Once the initialization is done, the value of each feature flag is guaranteed
+ * to be constant within the page.
+ * The values are also cached in SessionStorage, so that even after page
+ * refreshing, the values stay the same, unless:
+ *   - the cache TTL of 12 hours has been reached, or
+ *   - the current account is different than the account in use when the values
+ *     are loaded, i.e. a different session id is present in the cookies.
+ * In such cases, the values will be re-initialized, they may be changed.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { ClientContext, ClientContextObjectFactory } from
+  'domain/platform_feature/client-context-object.factory';
+import {
+  FeatureStatusSummary,
+  FeatureStatusSummaryObjectFactory,
+  FeatureSummaryDict
+} from 'domain/platform_feature/feature-status-summary-object.factory';
+import { PlatformFeatureBackendApiService } from
+  'domain/platform_feature/platform-feature-backend-api.service';
+import { I18nLanguageCodeService } from 'services/i18n-language-code.service';
+import { LoggerService } from 'services/contextual/logger.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { UrlService } from './contextual/url.service';
+import { BrowserCheckerService } from
+  'domain/utilities/browser-checker.service';
+
+interface FeatureFlagsCacheItem {
+  timestamp: number;
+  sessionId: string;
+  featureStatusSummary: FeatureStatusSummary;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PlatformFeatureService {
+  private static SESSION_STORAGE_KEY = 'SAVED_FEATURE_FLAGS';
+  private static SESSION_STORAGE_CACHE_TTL = 12 * 3600 * 1000; // 12 hours
+
+  private static COOKIE_NAME_FOR_SESSION_ID = 'SACSID';
+  private static COOKIE_NAME_FOR_SESSION_ID_IN_DEV = 'dev_appserver_login';
+
+  // TODO(#9154): Remove static when migration is complete.
+  static featureStatusSummary: FeatureStatusSummary = null;
```

Comment:
@MegrezZhu static is no longer required. Apologies for the inconvenience as I didn't close the issue yet.

Reply:
Yes! This happened because people used the service instance from UpgradedServices and not the one angular provided. This is not the case anymore. We have a way to use the service instance provided by angular without running into the angular injector error.